tools/expotools/src/publish-packages/tasks.ts

```diff
@@ -0,0 +1,729 @@
+      acc[pkg.packageName] = pkg;
+      return acc;
+    }, {});
+
+    // Verify that provided package names are valid.
+    for (const packageName of packageNames) {
+      if (!allPackagesObj[packageName]) {
+        throw new Error(`Package with provided name ${green(packageName)} does not exist.`);
+      }
+    }
+
+    const filteredPackages = allPackages.filter((pkg) => {
+      const isPrivate = pkg.packageJson.private;
+      const isIncluded = packageNames.length === 0 || packageNames.includes(pkg.packageName);
+      return !isPrivate && isIncluded && !IGNORED_PACKAGES.includes(pkg.packageName);
+    });
+
+    parcels.push(...(await Promise.all(filteredPackages.map(createParcelAsync))));
+
+    if (packageNames.length > 0) {
+      // Even if some packages have been explicitly listed as command arguments,
+      // we also must take their dependencies into account.
+
+      const parcelsObj = parcels.reduce((acc, parcel) => {
+        acc[parcel.pkg.packageName] = parcel;
+        return acc;
+      }, {});
+
+      await recursivelyResolveDependenciesAsync(allPackagesObj, parcelsObj, parcels);
+    }
+  }
+);
+
+/**
+ * Checks packages integrity and warns about violations.
+ * Integrity is violated if the current version of a package:
+ * - has no `gitHead` property in its package view.
+ * - commit to which `gitHead` refers is not an ancestor of the current head commit.
+ * - mismatches last version found in changelog.
+ */
+export const checkPackagesIntegrity = new Task<TaskArgs>(
+  {
+    name: 'checkPackagesIntegrity',
+    dependsOn: [prepareParcels],
+  },
+  async (parcels: Parcel[], options: CommandOptions): Promise<void | symbol> => {
+    logger.info('\n👁  Checking packages integrity...');
+
+    const resolver = async ({ pkg, pkgView, changelog }: Parcel): Promise<boolean> => {
+      if (!pkgView) {
+        // If package view is not there, then the package hasn't been released yet - no need to check integrity.
+        return true;
+      }
+
+      const isAncestor = !!pkgView.gitHead && (await Git.isAncestorAsync(pkgView.gitHead));
+      const lastChangelogVersion = await changelog.getLastPublishedVersionAsync();
+      const isVersionMatching = !lastChangelogVersion || pkgView.version === lastChangelogVersion;
+      const integral = isAncestor && isVersionMatching;
+
+      if (!integral) {
+        logger.warn(`\n⚠️  Integrity checks failed for ${green(pkg.packageName)}.`);
+      }
+      if (!pkgView.gitHead) {
+        logger.warn(`   Cannot find ${blue('gitHead')} in package view.`);
+      } else if (!isAncestor) {
+        logger.warn(
+          `   Local version ${cyan(pkgView.version)} has been published from different branch.`
+        );
+      }
+      if (!isVersionMatching) {
+        logger.warn(
+          `   Last version in changelog ${cyan(lastChangelogVersion!)}`,
+          `doesn't match local version ${cyan(pkgView.version)}.`
+        );
```

Comment:
With one argument this line would be wrapped by prettier as it's too long – and imho wrapping template strings doesn't look pretty well, it would look like this:

```typescript
logger.warn(
  `   Last version in changelog ${cyan(
    lastChangelogVersion!
  )} doesn't match local version ${cyan(pkgView.version)}.`
);
```
😅